### ISSUE TITLE
Why is CSSCRIPT_DIR necessary?

### ISSUE THREAD

**Issue by rconde01** [oleg-shilo/cs-script#90]
Why can't cscs just find it's own location?

I want to install cs-script through a conan package, but i need to have reference to an inc directory. It's not very straightfoward to add an environment variable when installing the package (since it wouldn't affect the current session). Relative directories in SearchDirs appears to be relative to the current directory. So it would be nice if I could do something like:

<SearchDirs>%CSSCRIPT_EXE_DIR%\inc<SearchDirs>

Now thinking a bit more, after installation i could do something like:
cscs.exe -config:set:SearchDirs=add:%CD%\inc

which is not too bad, but still it seems unnecessary

in fact, it seems RoslynDir is already interpreted as relative to cscs.exe...no?

**Comment by oleg-shilo** [comment]
I think there is a small confusion about the nature and objective of CSSCRIPT_DIR. 

It is not used to find CS-Script dependencies (e.g. include dirs, Roslyn) but to let third-party products who depend on CS-Script to find its system wide install location.

Thus Notepad++ uses globally installed cscs.exe instead of embedded one if it detects that CSSCRIPT_DIR is set to the valid path. 

Another point,  %CSSCRIPT_DIR%\inc is completely optional. You don't need to set it. In fact on Linux CS-Script is distribution is extremely minimalistic. The package consist of only a few binaries.

Roslyn is a completely different creature. It is CS-Script who depends on it so it needs to do everything to find it.

**Comment by rconde01** [comment]
Ok - that makes sense. But let me rephrase the question: Is there a way to specify search directories relative to cscs.exe? If not, it would be nice if you could.

**Comment by oleg-shilo** [comment]
Yes you can do this.

Just add to your script (or css_config.xml)  a probing dir that is based on `cscs_exe_dir` environment variable:
```C#
//css_dir %cscs_exe_dir%
//css_inc test.cs
```

The whole list of all envvars associated with CS-Script can be obtained with `cscs -help` command.

And of course in many cases you will be happy with the alternative approach when you can specify the script dependencies with respect to the script being executed. In this case you need to execute the script with `-l` switch. This will set the currentdir to the location of the script just before the execution.

**Comment by rconde01** [comment]
Perfect!
